### PR TITLE
Likes bookmark comment api integration

### PR DESCRIPTION
**Pull request by abhisheksahani-nova**
Note :-  Files to ignore 

1. All Backend files

2. All slice redux files - since its same as other slice file which have been reviewed earlier.

3. PostDropdown file - since its same dropdown as note app

Total approx lines to review -   480 lines approx

### CODE REVIEW

File: src/components/Comment/Comment.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  postUsername,
+  editCommentData,
+  setEditCommentData,
+}) {
+  const { _id, text, username, name, votes } = comment;
+
+  const [isCommentDropdownOpen, setIsCommentDropdownOpen] = useState(false);
+
+  const signInUser = localStorage.getItem("username");
+  const token = localStorage.getItem("token");
```
Comment (Dev-AkshatGupta) [inline src/components/Comment/Comment.jsx:22]: You can take them directly in the function instead

File: src/components/Post/Post.jsx [hunk 2]
```diff
@@ -1,46 +1,215 @@
...
+import { useDispatch, useSelector } from "react-redux";
+import TextareaAutosize from "react-textarea-autosize";
 
 function Post({ post, setIsPostEdit, setEditPostId }) {
   const { _id, content, username, name } = post;
+
+  const [isPostLikedBy, setIsPostLikedBy] = useState(false);
+  const [isPostBookmark, setIsPostBookmark] = useState(false);
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [showCommentBox, setShowCommentBox] = useState(false);
```
Comment (Dev-AkshatGupta) [inline src/components/Post/Post.jsx:23]: Can be done in one state which is having object instead  of boolean directly

**Review by Dev-AkshatGupta — APPROVED** [review]
Just few suggestion been given everything else seems quite good to me 
You are good to merge it

File: src/components/Post/Post.jsx [hunk 3]
```diff
@@ -1,46 +1,215 @@
...
+    );
+
+    setIsPostLikedBy(isLikedPost);
+  }, [postObj]);
+
+  useEffect(() => {
+    const isBookmark = bookmark.some((user) => user._id == _id);
+    setIsPostBookmark(isBookmark);
+  }, [bookmark]);
+
```
Comment (Vishnu-Aithal) [inline src/components/Post/Post.jsx:55]: suggestion - This approach is also fine, 
isLikedby and isBookMarkedBy can be just functions which return boolean by processing values from slice 
ex - 
const isLikedPost = () => {
    const thisPost = postObj.posts.filter((post) => post._id == _id);
    return thisPost[0].likes.likedBy.some(
        (user) => user.username == activeUsername
    );
};

you can use isLikedPost() wherever required.
it automatically reruns when component re-render when slice updates

**Review by Vishnu-Aithal — APPROVED** [review]
Code Looks Good , functionalites are working after login, Good To Merge

Few Suggestions you can implement later -
- instead of reading from local Storage on every render in your components you can store it in a new slice as currentUser and and load it in root App Component, then you can use everywhere else

File: src/components/Comment/Comment.jsx (hunk 1 above)
Comment (abhisheksahani-nova) [reply src/components/Comment/Comment.jsx:22]: Yaah thats pending , I will do that

File: src/components/Post/Post.jsx (hunk 2 above)
Comment (abhisheksahani-nova) [reply src/components/Post/Post.jsx:23]: Yupp actually realized later

File: src/components/Post/Post.jsx (hunk 3 above)
Comment (abhisheksahani-nova) [reply src/components/Post/Post.jsx:55]: Okay thanks , I will look into it

**Comment by abhisheksahani-nova** [comment]
> Code Looks Good , functionalites are working after login, Good To Merge
> 
> Few Suggestions you can implement later -
> 
> * instead of reading from local Storage on every render in your components you can store it in a new slice as currentUser and and load it in root App Component, then you can use everywhere else

Yaah thats pending I have to do that

File: src/components/Comment/Comment.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+              ></i>
+            )}
+          </div>
+          <small> {text} </small>
+        </div>
+
+        <div className="d-flex gap-1 comment-icon-container">
+          <div>
+            <i
+              class="fa-regular fa-thumbs-up comment-like-icon"
```
Comment (jalajcodes) [inline src/components/Comment/Comment.jsx:74]: should be className

**Review by jalajcodes — APPROVED** [review]

**Pull request merged**